src/Uno.UI/UI/Xaml/Media/PathFigure.cs

```diff
@@ -48,10 +53,20 @@ public PathSegmentCollection Segments
 				typeof(PathFigure),
 				new FrameworkPropertyMetadata(
 					defaultValue: null,
+					propertyChangedCallback: OnSegmentsChanged,
 					options: FrameworkPropertyMetadataOptions.ValueInheritsDataContext  | FrameworkPropertyMetadataOptions.LogicalChild | FrameworkPropertyMetadataOptions.AffectsMeasure
 				)
 			);
 
+		private static void OnSegmentsChanged(DependencyObject dependencyobject, DependencyPropertyChangedEventArgs args)
+		{
+			if (dependencyobject is PathFigure pf && args.NewValue is PathSegmentCollection collection)
+			{
+				// Ensure parent is set to allow Measure invalidation to pass through
+				collection.SetParent(pf);
```

Comment:
This should be achieved by the DP itself, no?

Reply:
Not sure if Carl added this after testing, or copy-pasting, or 'just in case', but it's fairly common to call SetParent() from controls and classes (eg https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/DependencyObjectCollection.cs#L58)